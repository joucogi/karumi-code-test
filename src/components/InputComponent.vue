<template>
  <b-form-input :type="type" :placeholder="placeholder"></b-form-input>
</template>
<script>
export default {
  props: {
    type: {
      type: String,
      default: "email",
      required: false
    },
    placeholder: {
      type: String,
      default: "password",
      required: false
    }
  }
};
</script>
<style scoped></style>
