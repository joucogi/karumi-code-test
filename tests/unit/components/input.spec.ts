import { shallowMount, Wrapper } from "@vue/test-utils";
import Vue from "vue";
import { BootstrapVue, BFormInput } from "bootstrap-vue";

import InputComponent from "@/components/InputComponent.vue";

Vue.use(BootstrapVue);

let wrapper: Wrapper<Vue>;

beforeEach(() => {
  wrapper = shallowMount(InputComponent);
});

describe("Input Component", () => {
  it("contents BFormInput component", () => {
    expect(wrapper.findComponent(BFormInput).exists()).toBeTruthy();
  });

  it("renders correctly", () => {
    expect(wrapper.element).toMatchSnapshot();
  });
});
