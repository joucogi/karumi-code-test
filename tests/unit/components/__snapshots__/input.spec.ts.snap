// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input Component renders correctly 1`] = `
<b-form-input-stub
  debounce="0"
  placeholder="password"
  type="email"
  value=""
/>
`;
